app/models/label/LabelValidationTable.scala

```diff
@@ -86,9 +99,143 @@ object LabelValidationTable {
     labelValidationId
   }
+
+  def getOwnValidationCounts: Map[String, Int] = db.withSession { implicit session =>
+    LabelValidationTable.labelValidationTable
+      .innerJoin(LabelTable.labels).on(_.labelId === _.labelId)
+      .innerJoin(AuditTaskTable.auditTasks).on(_._2.auditTaskId === _.auditTaskId)
+      .groupBy(_._2.userId).map { case (_userId, group) => (_userId, group.length) }.list.toMap // _2.userId is the userId from "audit_task"
+  }
+
+  def getOwnValidationAgreedCounts: Map[String, Int] = db.withSession { implicit session =>
+    LabelValidationTable.labelValidationTable.filter(_.validationResult === 1)
+      .innerJoin(LabelTable.labels).on(_.labelId === _.labelId)
+      .innerJoin(AuditTaskTable.auditTasks).on(_._2.auditTaskId === _.auditTaskId)
+      .groupBy(_._2.userId).map { case (_userId, group) => (_userId, group.length) }.list.toMap
+  }
+
+  /**
+    * Select validation counts per user.
+    *
+    * @return list of tuples of (labeler_id, validator_role, validation_result, count)
+    */
+  def getValidationCountsPerUser: List[(String, String, Int, Int)] = db.withSession { implicit session =>
+    val audits = for {
+      _validation <- labelValidationTable
+      _label <- labelsWithoutDeleted if _label.labelId === _validation.labelId
+      _audit <- auditTasks if _label.auditTaskId === _audit.auditTaskId
+      _user <- users if _user.username =!= "anonymous" && _user.userId === _audit.userId // User who placed the label
+      _validationUser <- users if _validationUser.username =!= "anonymous" && _validationUser.userId === _validation.userId // User who did the validation
+      _userRole <- userRoles if _validationUser.userId === _userRole.userId
+      _role <- roleTable if _userRole.roleId === _role.roleId
+    } yield (_user.userId, _role.role, _validation.labelId, _validation.validationResult)
+
+    // Counts the number of labels for each user by grouping by user_id and role.
+    audits.groupBy(l => (l._1, l._2, l._4)).map{ case ((uId, role, result), group) => (uId, role, result, group.length) }.list
+  }
+
+  /**
+    * Select validation counts per user, split into agreed and total
+    *
+    * @return map of labeler_id -> (validator_role, total, agreed)
+    */
+  def getCategorizedValidationCountsPerUser: scala.collection.mutable.HashMap[String, (String, Int, Int)] = db.withSession { implicit session =>
+    val validationCounts = getValidationCountsPerUser
+
+    // Map userId -> (role, total, agreed)
+    // Ignore role for now because we need to figure out what to do with the role
+    // There's 2 roles, the labeler and the validator, and it's unclear how to filter by "Include researchers"
+    val validations = new scala.collection.mutable.HashMap[String, (String, Int, Int)]
+
+    validationCounts.foreach{ x =>
+      var total = 0
+      var agreed = 0
+
+      val role = x._2
+
+      if (validations.contains(x._1)) {
```

Comment:
Since you have to use a tuple where you index using `._1`, `._2`, etc., it would be really helpful to have comments before code blocks like this explaining what's happening.